samza-core/src/main/scala/org/apache/samza/config/ShellCommandConfig.scala

```diff
@@ -25,6 +25,11 @@ object ShellCommandConfig {
    */
   val ENV_COORDINATOR_SYSTEM_CONFIG = "SAMZA_COORDINATOR_SYSTEM_CONFIG"
 
+  /**
+   * This environment variable is used to pass a JSON serialized map of configs provided during job submission.
+   */
+  val ENV_SUBMISSION_CONFIG = "SAMZA_SUBMISSION_CONFIG"
```

Comment:
where is it used?

Reply:
This will be used in YarnJob when submitting the job to Yarn. This environment variable will encapsulate config overrides users would like to override on AM.

Reply:
So code using it will be added later? then it is fine. In general it is easier to read PR if all relevant code is in. But this is fine too.

Reply:
Good point, yes, it will be added later. I am splitting them to make PRs relatively small.